### PR TITLE
Angular cleanup: Remove deprecated methods from util_srv

### PR DESCRIPTION
**Pull request by ashharrison90**
**What this PR does / why we need it**:

- removes deprecated methods `showModal` and `hideModal` from `util_srv`
- removes angular injections
- updates `keybindingSrv` to use the new events instead

**Which issue(s) this PR fixes**:

For https://github.com/grafana/grafana/issues/35326

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/core/services/util_srv.ts [hunk 1]
```diff
@@ -55,50 +51,6 @@ export class UtilSrv {
...
-    deprecationWarning('UtilSrv', 'hideModal', 'showModalReact');
-    if (this.modalScope && this.modalScope.dismiss) {
-      this.modalScope.dismiss();
-    }
-  }
-
-  /**
-   * @deprecated use showModalReact instead
-   */
-  showModal(options: any) {
```
Comment (dprokop) [inline public/app/core/services/util_srv.ts:71]: Removing this needs to be clearly communicated to the plugins communitu, as there may be angular panels that that still use this API. cc @mckn @jackw

Comment (ashharrison90) [reply public/app/core/services/util_srv.ts:71]: ah gotcha. didn't think about plugins! so are we even in a position to do the cleanup in this file yet, or do we wait a bit for 9.0 or something? 🤔

Comment (jackw) [reply public/app/core/services/util_srv.ts:71]: Sorry @ashharrison90 but given these are breaking changes I think we've gonna need to wait for a major release for this work.

Comment (ashharrison90) [reply public/app/core/services/util_srv.ts:71]: that's ok, makes sense 👍  i'll close for now, roll on 9.0! 🚀

**Pull request closed without merging**